port/unix/omrsysinfo.c

```diff
@@ -3568,18 +3585,31 @@ omrsysinfo_startup(struct OMRPortLibrary *portLibrary)
 	 */
 	if (0 == attachedPortLibraries) {
 		if (omrthread_monitor_init_with_name(&cgroupEntryListMonitor, 0, "cgroup entry list monitor")) {
-			return OMRPORT_ERROR_STARTUP_SYSINFO_MONITOR_INIT;
+			rc = OMRPORT_ERROR_STARTUP_SYSINFO_MONITOR_INIT;
+			goto _end;
 		}
 	}
 	attachedPortLibraries += 1;
-	isRunningInContainer(portLibrary, &PPG_isRunningInContainer);
-#endif /* defined(LINUX) */
-	return 0;
 
-#if (defined(S390) || defined(J9ZOS390))
-	PPG_stfleCache.lastDoubleWord = -1;
-#endif
```

Comment:
Have you investigated whether this code should have been inserted earlier (so it isn't dead) as opposed to just deleting it?

Reply:
Seems like it is still needed. However I believe its usage in omrsysinfo is incorrect: https://github.com/eclipse/omr/blob/923ae683b6b7db392aa9e9cb9e3d2a4e4f9bd20a/port/unix/omrsysinfo.c#L1105-L1110

Whereas it is initialized to -1 in `omrsysinfo_startup`. In j9sysinfo it is correct: https://github.com/eclipse-openj9/openj9/blob/da8311db8c6efa4cdabce83c4070fafb2d78d6be/runtime/port/unix/j9sysinfo.c#L707-L712
Should a new PR be opened to fix these? Maybe I should readd the dead code and a new PR created to move the dead code and correct the check.

Reply:
This change was done as part of https://github.com/eclipse/omr/pull/4503 where we migrated processor/feature detection from OpenJ9 to OMR. 

In OpenJ9, the following is done:
```
int32_t
j9sysinfo_startup(struct J9PortLibrary *portLibrary)
{
#if (defined(S390) || defined(J9ZOS390))
	PPG_stfleCache.lastDoubleWord = -1; <--- INIT to -1
#endif

...

static BOOLEAN
testSTFLE(struct J9PortLibrary *portLibrary, uint64_t stfleBit)
{
        ...
	/* If it is the first time, read stfle and cache it */
	if (-1 == *stfleRead) { <--- COMPARED TO -1
		*stfleRead = getstfle(LAST_DOUBLE_WORD, (uint64_t*)mem);
	}
```

In OMR, `PPG_stfleCache.lastDoubleWord` is set to 0 due to the dead code in `omrsysinfo_startup`. In `omrsysinfo_test_stfle`, we initialize it if its the first time (compare it to 0):
```
static BOOLEAN
omrsysinfo_test_stfle(struct OMRPortLibrary *portLibrary, uint64_t stfleBit)
{
        ...
	/* If it is the first time, read stfle and cache it */
	if (0 == *stfleRead) { <--- INIT TO 0 and COMPARED TO 0
		*stfleRead = getstfle(LAST_DOUBLE_WORD, (uint64_t*)mem);
	}
```

If we move the dead code earlier, `PPG_stfleCache.lastDoubleWord` will be set to `-1` and it won't be correctly initialized in `omrsysinfo_test_stfle`. Removing the dead code is the correct approach. It should be handled in a separate issue/PR and not mixed with cgroup work.

On another note, `lastDoubleWord` is of type `uintptr_t`. Writing a negative value (-1) in `PPG_stfleCache.lastDoubleWord` is incorrect on the OpenJ9 side.